src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1669,6 +2505,177 @@ function scale_edge_length(angleEdgeHere, u, v)
+         ! OR is adjacent to an inactive floating margin cell
+
+         ! Check if neighboring cells contain ice and have bed topo below sea level
+         nEmptyNeighbors = 0
+         do iEdge = 1, nEdgesOnCell(iCell)
+           iNeighbor = cellsOnCell(iEdge, iCell)
+           if ( (((thickness(iNeighbor) == 0.0_RKIND) &
+              .and. bedTopography(iNeighbor) < config_sea_level)) &
+              .or. ((li_mask_is_floating_ice(cellMask(iNeighbor)) &
+              .and. li_mask_is_margin(cellMask(iNeighbor)) &
+              .and. (.not. li_mask_is_dynamic_ice(cellMask(iNeighbor)))))) then
+               nEmptyNeighbors = nEmptyNeighbors + 1
+           endif
+         enddo
+
+         if ( nEmptyNeighbors > 0 &
+           .and. li_mask_is_grounded_ice(cellMask(iCell)) &
+           .and. bedTopography(iCell) < config_sea_level &
+           .and. li_mask_is_dynamic_ice(cellMask(iCell)) ) then
+
+            groundedMarineMarginMask(iCell) = 1
+         else
+            groundedMarineMarginMask(iCell) = 0
+         endif
+      enddo
+
+      ! The calving volume needs to be distributed in three ways:
+      ! 1. We need to first remove any "thin" ice in front of this cell
+      ! 2. Then we remove ice from this cell if damage > threshold
+      do iCell = 1, nCells
+         if ( ( (li_mask_is_floating_ice(cellMask(iCell)) .and. li_mask_is_dynamic_ice(cellMask(iCell)) ) & ! floating dynamic
+                .or.  (groundedMarineMarginMask(iCell) == 1) ) &
+              .and. li_mask_is_margin(cellMask(iCell)) &
+              .and. (damage(iCell) .ge. config_damage_calving_threshold)) then
+
+            ! First remove ice from "thin" neighbors
+            do iNeighbor = 1, nEdgesOnCell(iCell)
+               jCell = cellsOnCell(iNeighbor, iCell)
+               if (li_mask_is_floating_ice(cellMask(jCell)) .and. .not. li_mask_is_dynamic_ice(cellMask(jCell))) then
+                  ! this is a thin neighbor - remove the whole cell volume
+                  removeVolumeHere = cellVolume(jCell) ! how much we want to remove here
+                  calvingThickness(jCell) = removeVolumeHere / areaCell(jCell)
+                      ! < apply to the field that will be used, in thickness units
+                  cellVolume(jCell) = cellVolume(jCell) - removeVolumeHere ! update accounting on cell volume
+               endif
+            enddo
+
+            ! Now remove ice from iCell
+            removeVolumeHere = cellVolume(iCell)
+            ! remove the whole damaged cell
+            calvingThickness(iCell) = removeVolumeHere / areaCell(iCell)
+            ! < apply to the field that will be used in thickness units
+            cellVolume(iCell) = cellVolume(iCell) - removeVolumeHere 
+            ! update accounting on cell volume
+
+            ! Now calved away the neighboring cells if the damage is also > the threshold value
+            ! I choose to disable this section (i.e., no recursive calving) -- TZ
```

Comment:
I don't think we want to use the threshold method without a way to remove all ice over the threshold that is contiguous with the calving front.  We have code elsewhere that would do this kind of flood fill operation.  I propose we add TODO notes in this routine and make a note in the Registry that this method is still in development.  And then we work on fixing it in a follow up merge after this one is done.

Reply:
Right, I agree with it. However, we should note that if we apply the flood fill algorithm for this threshold method, the calving front will retreat even much faster. A TODO comment line is added